test/e2e/kubelet.go

```diff
@@ -92,6 +97,22 @@ func waitTillNPodsRunningOnNodes(c clientset.Interface, nodeNames sets.String, p
 	})
 }
 
+// waitForKubeletCondition waits for the passed-in condition function to return true,
+// or the timeout to expire.
+func waitForKubeletCondition(timeout, poll time.Duration, cond kubeletCondition) error {
+
+	framework.Logf("Waiting up to %v for kubelet's host to be in expected condition", timeout)
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(poll) {
+		done, err := cond()
+		if done {
+			return err
+		}
+		framework.Logf("Waiting another %v for host to be in expected condition", poll)
+	}
+
+	return fmt.Errorf("gave up after waiting %v for kubelet's host to be in expected condition", timeout)
+}
```

Comment:
@gnufied I added [Slow] to kubelet.go -- good suggestion and I meant to do that but forgot. I think, as of now, all of the pv related e2e tests are slow, some with the possibility of 5m waits. I have not added [Slow] to the test in volumes.go since I did not write them and have not monitored their speed.

I will use an existing pattern for the wait loop.